test/Tests/src/Data/Text_Spec.enso

```diff
@@ -339,6 +341,33 @@ spec =
             kshi_chars = [2325, 2381, 2359, 2367]
             Text.from_utf_16 kshi_chars . should_equal kshi
 
+        Test.specify "should insert text at a positive index position" <|
+            hello_world.insert 0 cruel . should_equal " CruelHello World!"
+            hello_world.insert 5 cruel . should_equal "Hello Cruel World!"
+            hello_world.insert (hello_world.length - 1) cruel . should_equal "Hello World Cruel!"
+            hello_world.insert hello_world.length cruel . should_equal "Hello World! Cruel"
+            txt = kshi + facepalm + accent_1
+            txt.insert 1 cruel . should_equal (kshi + cruel + facepalm + accent_1)
+            txt.insert 3 cruel . should_equal (kshi + facepalm + accent_1 + cruel)
```

Comment:
I think it may be worth adding the cases with 0 and 2 here too. Probably not absolutely necessary, but low cost for being more comprehensive.

Reply:
Sure

Reply:
done

Reply:
Ideally we would use something like property testing since writing all those cases is a bit annoying. But that's for future

Reply:
True, I think it would be great to have this, but probably too much upfront cost for right now.